src/main/java/com/example/hw18/controller/Controller.java

```
@@ -0,0 +1,143 @@
+        return user;
+    }
+
+    @PutMapping("/newCar")
+    public Car createCar(@RequestBody Car car) {
+
+        for (Car car1 : cars) {
+            if (car.getCarId() == car1.getCarId()) {
+                throw new ResponseStatusException(
+                        HttpStatus.CONFLICT, "car with this id is arleady exist");
+            }
+        }
+        cars.add(car);
+
+        return car;
+    }
+
+    @PostMapping("/makeOvners/{userId}/{carId}")
+    public User makeOvners(@PathVariable(value = "userId") int userId,
+                           @PathVariable(value = "carId") int carId) {
+
+        garage.get(userService.findUserById(garage, userId)).add(cars.get(carId));
+
+        for (Car car : garage.get(userService.findUserById(garage, userId))) {
+            if (car.getCarId() == carId) {
+                car.setUserIdOvner(userId);
+            }
+        }
+        return userService.findUserById(garage, userId);
+    }
+
+    @PostMapping("/deleteCar/{carId}")
+    public List<Car> deleteCar(@PathVariable(value = "carId") int carId) {
+        cars.removeIf(car -> car.getCarId() == carId);
+        return cars;
+    }
+
+    @PostMapping("/deleteUser/{userId}")
+    public User deleteUser(@PathVariable(value = "userId") int userId) {
+        for (User user : garage.keySet()) {
+            if (user.getUserId() == userId)
+                garage.remove(user);
+            return user;
+        }
+        return null;
+    }
+
+    @PostMapping("/updateUser/{userId}")
+    public User updateUser(@RequestBody User incomingData,
+                           @PathVariable(value = "userId") int userId) {
+        for (Map.Entry<User, ArrayList<Car>> item : garage.entrySet()) {
+            if (item.getKey().equals(userService.findUserById(garage, userId))) {
+                item.getKey().setName(incomingData.getName());
+            }
+            return item.getKey();
+        }
+        return null;
```

in this case, you need to throw exception (object not found)